src/Mod/Part/Gui/SectionCutting.cpp

```diff
@@ -603,7 +603,7 @@ void SectionCut::startCutting(bool isInitial)
 
     // store color and transparency of first object
     App::Color cutColor;
-    int cutTransparency;
+    int cutTransparency {-1};

```

Comment:
Why not
`int cutTransparency = 0;`

I think we should initialize to sensible values just in case the initialization will be used in an unforeseen case.

Reply:
My reasoning was that if that variable is ever used without getting set, something has gone wrong. Initializing it to zero , which is a valid value, would mask the error when debugging. If you are not concerned with that eventuality I will adjust the PR.

Reply:
I don't understand the problem at all because few lines below the variable is initialized by reading the transparency of an object. We already check if that object exists.
If the valid object has for whatever reason no transparency, the int would have the value 0. So we can safely initialize it as o.
However, what I learned from C++ is that a new int gets internally already the value 0. Therefore no special initialization is necessary if 0 is a usable value.

Reply:
> However, what I learned from C++ is that a new int gets internally already the value 0. Therefore no special initialization is necessary if 0 is a usable value.

That is not true: the compiler is not required to initialize POD types unless they have static storage duration, and when compiled in Release mode usually does not do so. Because of this subtlety it is widely regarded as a best practice to initialize all POD variables in C++. See, e.g. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es20-always-initialize-an-object

Reply:
Thanks. so I will in future initialize these variables.

Here I would prefer to initialize to 0.

Reply:
OK.